drivers/soundwire/generic_bandwidth_allocation.c

```diff
@@ -131,96 +138,126 @@ static void sdw_compute_master_ports(struct sdw_master_runtime *m_rt,
 		port_bo += bps * ch;
 	}
 
+	t_data.lane = params->lane;
 	sdw_compute_slave_ports(m_rt, &t_data);
 }
 
 static void _sdw_compute_port_params(struct sdw_bus *bus,
 				     struct sdw_group_params *params, int count)
 {
 	struct sdw_master_runtime *m_rt;
-	int hstop = bus->params.col - 1;
-	int port_bo, i;
+	int port_bo, i, l;
+	int hstop;
 
 	/* Run loop for all groups to compute transport parameters */
-	for (i = 0; i < count; i++) {
-		port_bo = 1;
+	for (l = 0; l < SDW_MAX_LANES; l++) {
+		/* reset hstop for each lane */
+		hstop = bus->params.col - 1;
+		for (i = 0; i < count; i++) {
```

Comment:
what does 'count' represent? I can't recall what this was used for.

Reply:
'count' is the number of group params.

Reply:
i guess I was never clear on what the 'group' represents...

Reply:
I think the basic idea is to group the streams with the same rates together. Then calculate the required bandwidth etc. And the commit is to add lane into sdw_group_params{} because we should use different parameters for different lanes.